wallet-new/test/MarshallingSpec.hs

```diff
@@ -5,28 +5,62 @@ import           Universum
 import           Cardano.Wallet.API.V1.Types
 import           Cardano.Wallet.Orphans ()
 import           Data.Aeson
-import           Pos.Util.BackupPhrase (BackupPhrase)
+import           Data.Typeable (typeRep)
+import           Pos.Crypto (PassPhrase, emptyPassphrase)
 import           Test.Hspec
 import           Test.Hspec.QuickCheck
-import           Test.QuickCheck
+import           Test.QuickCheck (Arbitrary, arbitrary, forAll, (===))
 import           Test.QuickCheck.Instances ()
+import           Test.QuickCheck.Property (Property, failed, property, reason)
+
+import           Pos.Util.BackupPhrase (BackupPhrase)
 
 import qualified Pos.Core.Types as Core
 
 -- | Tests whether or not some instances (JSON, Bi, etc) roundtrips.
 spec :: Spec
 spec = describe "Marshalling & Unmarshalling" $ do
-  prop "Aeson BackupPhrase roundtrips" (aesonRoundtrip @BackupPhrase Proxy)
-  prop "Aeson AssuranceLevel roundtrips" (aesonRoundtrip @AssuranceLevel Proxy)
-  prop "Aeson Payment roundtrips" (aesonRoundtrip @Payment Proxy)
-  prop "Aeson PaymentDistribution roundtrips" (aesonRoundtrip @PaymentDistribution Proxy)
-  prop "Aeson NewWallet roundtrips" (aesonRoundtrip @NewWallet Proxy)
-  prop "Aeson Coin roundtrips" (aesonRoundtrip @Core.Coin Proxy)
-  prop "Aeson TransactionGroupingPolicy roundtrips" (aesonRoundtrip @TransactionGroupingPolicy Proxy)
-  prop "Aeson TransactionType roundtrips" (aesonRoundtrip @TransactionType Proxy)
-  prop "Aeson Transaction roundtrips" (aesonRoundtrip @Transaction Proxy)
+    describe "Roundtrips" $ do
+        aesonRoundtripProp @BackupPhrase Proxy
+        aesonRoundtripProp @AssuranceLevel Proxy
+        aesonRoundtripProp @Payment Proxy
+        aesonRoundtripProp @PaymentDistribution Proxy
+        aesonRoundtripProp @NewWallet Proxy
+        aesonRoundtripProp @Core.Coin Proxy
+        aesonRoundtripProp @TransactionGroupingPolicy Proxy
+        aesonRoundtripProp @TransactionType Proxy
+        aesonRoundtripProp @Transaction Proxy
+
+    describe "Special" $ do
+        describe "password" $ do
+            prop "empty" $
```

Comment:
"empty" is a bit generic. What about `prop "empty string decodes to empty passphrase"`? Same applies for the rest of the two functions.

Reply:
Also, there doesn't seem to be properties in the QuickCheck sense, but more unit tests, so I would suggest change `prop` to simply `it`. What do you think?

Reply:
Would be nice, I get used more to it